demos/dotnet/dotnet/dotnet.cs

```diff
@@ -293,20 +312,39 @@ public static ProjectProperties InitializeProperties(string[] args, Log log, boo
             properties.AssemblyName = Path.GetFileNameWithoutExtension(properties.Sources[0]);
         }
 
+        string result = string.Empty;
+
+        var osName = GetOSName(Environment.OSVersion);
```

Comment:
Why do you read the OS from environment? How will people compile for Win7 on Win 10, for example? Also, some of the runtime packages don't have folders for all OSes. And so the tool will fail. I thought we decided to skip OS specific builds till we have packages that support Linux and/or Mac.